### PR TITLE
update pull-request.yml to ignore github actions when label "ignore-protected-files-pr" is added

### LINKED ISSUE #2617: API: Create PR GitHub Actions logic for the `ignore-protected-files-pr` label
**Is your feature request related to a problem? Please describe.**

1. The `pull-request.yml` workflow file has a section called `Check-Sensitive-Files`
1. This GitHub action:
    1. was created to flag PRs that edited sensitive files that we want to protect from unauthorized changes
    1. is run on all PRs except those submitted by `dependabot`
1. There are cases where community members need to edit sensitive files. When this is done, the `Check-Sensitive-Files` test fails. Our formal GitHub contributors can only merge PRs where all tests pass. This means the Repository owners have to merge the PRs. This creates a bottleneck in PR reviews

**Describe the solution you'd like**

1. There is a `ignore-sensitive-files-pr` label in this repository
1. We want only the `Check-Sensitive-Files` GitHub action to be ignored if this label is applied to the PR by any of our GitHub contributors
1. No other functionality must be altered, especially the logic where the  `Check-Sensitive-Files` GitHub action is also not run when the PR author is the `dependabot` user.

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- See above

**Additional context**
- N/A

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by govindsingh3477**
**What kind of change does this PR introduce?**

feature

**Issue Number:**

Fixes #2617

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**
There are cases where community members need to edit sensitive files. When this is done, the Check-Sensitive-Files test fails. The formal GitHub contributors can only merge PRs where all tests pass. This means the Repository owners have to merge the PRs. This creates a bottleneck in PR reviews. updated pull-request.yml file for GitHub action to be ignored if this label is applied to the PR by any of our GitHub contributors

**Does this PR introduce a breaking change?**

No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

YEs

## Summary by CodeRabbit

- **Chores**
	- Updated the GitHub Actions workflow to refine the criteria for unauthorized file change checks.
	- Improved comments and formatting for better clarity.

### CODE REVIEW

**Comment by palisadoes** [comment]
@govindsingh3477 

- This is failing when the label is applied. I reran all the checks to test.

**Comment by palisadoes** [comment]
Please work with @varshith257 to figure out a solution

File: .github/workflows/pull-request.yml
Comment (varshith257) [inline .github/workflows/pull-request.yml:94]:
```suggestion
    if: ${{ github.actor != 'dependabot[bot]' && !contains(github.event.pull_request.labels.*.name, 'ignore-sensitive-files-pr') }}
```

**Review by varshith257 — CHANGES_REQUESTED** [review]

**Review by varshith257 — APPROVED** [review]

**Comment by palisadoes** [comment]
@varshith257 Are you able to merge this?

**Comment by varshith257** [comment]
@palisadoes No

image

**Comment by palisadoes** [comment]
Please refresh the screen to verify

**Comment by varshith257** [comment]
@palisadoes I think permissions aren't sufficient to merge

**Comment by palisadoes** [comment]
Ask one of the other members to approve to check for sure.

**Comment by palisadoes** [comment]
The develop branches require two approvals.

**Review by noman2002 — APPROVED** [review]

**Comment by varshith257** [comment]
@noman2002 We are just checking the community members can merge PRS :)

**Comment by noman2002** [comment]
@varshith257 I think there was changes in unauthorised files and only admin can merge those PRs. May be that is why its not appearing.

**Comment by varshith257** [comment]
Yes, for that we have enabled skipping that step with the label `ignore-sensitive-files-pr`

**Comment by palisadoes** [comment]
@noman2002 It was because it requires two approvals for the `develop` branch.
@varshith257 Please work with @govindsingh3477 to open a similar issue in `talawa-admin`. Then you and another member should approve the work. Let's see if non owners can then merge. The rules allow merges for >= 2 approvals

**Comment by palisadoes** [comment]
@noman2002 I'm trying to see how we can delegate some of the PR approvals.

**Comment by varshith257** [comment]
@palisadoes Yes

**Comment by palisadoes** [comment]
@varshith257 It needs to be approvals from formal GitHub members like you, @AVtheking, @gautam-divyanshu, @pranshugupta54

**Comment by palisadoes** [comment]
Please try again.

**Comment by varshith257** [comment]
Yes!  wait let us simulate the same and will tag you

**Comment by palisadoes** [comment]
I'll be sleeping. Please work with @noman2002

**Pull request merged**
